src/web-ui/src/router/index.js

```diff
@@ -83,7 +83,7 @@ AmplifyEventBus.$on('authState', async (state) => {
       console.log('store user does not exist for cognito user... creating on the fly')
       let identityId = credentials ? credentials.identityId : null;
       let provisionalUserId = AmplifyStore.getters.personalizeUserID;
-      const { data } = await UsersRepository.createUser(provisionalUserId, cognitoUser.username, cognitoUser.attributes.email, identityId)
+      const { data } = await UsersRepository.createUser(provisionalUserId, cognitoUser.attributes.email, cognitoUser.attributes.email, identityId)
```

Comment:
This change creates tech debt. Since the username field is going away, the `createUser` function for the repository and the API call to the create the user in the Users service should change to no longer take a username.

Reply:
After discussing with the team about the tradeoffs of having the username field separately, we would like to avoid the risk of impacting the existing functionality and workshop or creating tech debt for the future.